02-wrangling.Rmd

```diff
@@ -50,23 +49,34 @@ and the columns are the variables:
 knitr::include_graphics("img/vars.jpeg")
 ```
 
-From a computer programming perspective, in R, a data frame is a special subtype of a list object whose elements (columns) are *vectors*. 
-For example, the data frame below has 3 elements that are vectors whose names are `state`, `year` and `population`.
+From a computer programming perspective, in R, a data frame is a special subtype of a list object whose elements (columns) are *vectors*. For example, the data frame below has three vectors whose names are `state`, `year` and `population`.
 
 ```{r 02-vectors, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "Data frame with 3 vectors", fig.retina = 2, out.width = "650"}
 knitr::include_graphics("img/vectors.jpeg")
 ```
 
 ### What is a vector?
 
-In R, vectors are objects that can contain 1 or more elements. The vector elements are ordered, and they must all be of the same type. Common 
-example types of vectors are character (e.g., letter or words), numeric (whole numbers and fractions) and logical (e.g., `TRUE` or `FALSE`). In 
-the vector shown below, the elements are of numeric type:
+In R, **vectors** are objects that can contain one or more elements. The vector elements are ordered, and they must all be of the same *data type*. R has several different basic data types, as shown in table \@ref(tab:datatype-table).
+
+In the vector shown below, the elements are of numeric type:
 
 ```{r 02-vector, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "Example of a numeric type vector", fig.retina = 2, out.width = "600"}
 knitr::include_graphics("img/vector.jpeg")
 ```
 
+
+Table: (\#tab:datatype-table) Basic data types in R
+
+| Data type | Description | Example |
+|-------|------------------|-----------|
+| character | Letters or numbers surrounded by quotes |  "1", "Hello world!" |
+| numeric | Whole numbers and numbers with decimal values |   1.23, 20 |  
```

Comment:
Integer is numeric. Numeric is a higher umbrella that includes integer and double. We probably want to state that here. And we probably want to tell them that 99% of the time if they are working with numbers, they will be double. That is the default when you create a vector using `c` from numbers in R (even whole numbers), and when you read in whole numbers via `read_csv` (or I am at least 99% sure about that - please confirm).

Reply:
Okay I introduced the idea of factors here and then I mention that we will discuss them more in depth in Ch 4!